core/DataTable.php

```diff
@@ -573,7 +573,13 @@ public function addDataTable(DataTable $tableToSum)
             $this->aggregateRowFromSimpleTable($row);
         } else {
             $columnAggregationOps = $this->getMetadata(self::COLUMN_AGGREGATION_OPS_METADATA_NAME);
-            foreach ($tableToSum->getRows() as $row) {
+            foreach ($tableToSum->getRowsWithoutSummaryRow() as $row) {
+                $this->aggregateRowWithLabel($row, $columnAggregationOps);
+            }
+            // we do not use getRows() as this method might get called 100k times when aggregating many datatables and
+            // this takes a lot of time.
+            $row = $tableToSum->getRowFromId(DataTable::ID_SUMMARY_ROW);
+            if ($row) {
```

Comment:
Quick note: It might be not as easy. Eg `getFirstRow` will be more complicated as it should not return the SummaryRow as a first row if there are other rows. Also `getLastRow` should return the summary row first. Those kinda implementations will get a bit more complicated.

Also sort etc. might get a bit harder since we shouldn't sort the summary row. Here he would have to create a new `$rows` array without the summary row.